pennylane/ops/op_math/exp.py

```diff
@@ -147,14 +146,35 @@ def __repr__(self):
             else f"Exp({self.coeff} {self.base.name})"
         )
 
+    # pylint: disable=attribute-defined-outside-init
+    def __copy__(self):
+        # this method needs to be overwritten because the base must be copied too.
+        copied_op = object.__new__(type(self))
+        # copied_op must maintain inheritance structure of self
+        # Relevant for symbolic ops that mix in operation-specific components.
+
+        for attr, value in vars(self).items():
+            if attr not in {"_hyperparameters"}:
+                setattr(copied_op, attr, value)
+
+        copied_op._hyperparameters = copy(self.hyperparameters)
+        copied_op.hyperparameters["base"] = copy(self.base)
+        copied_op._data = copy(self._data)
+
+        return copied_op
+
+    @property
+    def hash(self):
+        return hash((str(self.name), self.base.hash, self.coeff))
```

Comment:
This raises an error for TensorFlow and JAX because `self.coeff` is a `'tensorflow.python.framework.ops.EagerTensor'` and `'jaxlib.xla_extension.DeviceArray'` respectively. How should that be handled?